Improve startup of K8S tests

When K8S tests start, sometimes, very rarely, example DAGs are not parsed on time for the first unit tests to run. 

This results in (example https://github.com/apache/airflow/actions/runs/11157751788/job/31013822370#step:9:1497)


```python
_______________ TestKubernetesExecutor.test_integration_run_dag ________________
  
  self = <kubernetes_tests.test_kubernetes_executor.TestKubernetesExecutor object at 0x7f700fbdd100>
  
      @pytest.mark.execution_timeout(300)
      def test_integration_run_dag(self):
          dag_id = "example_kubernetes_executor"
  >       dag_run_id, execution_date = self.start_job_in_kubernetes(dag_id, self.host)
  
  test_kubernetes_executor.py:31: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  test_base.py:250: in start_job_in_kubernetes
      result_json = self.start_dag(dag_id=dag_id, host=host)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <kubernetes_tests.test_kubernetes_executor.TestKubernetesExecutor object at 0x7f700fbdd100>
  dag_id = 'example_kubernetes_executor', host = 'localhost:27399'
  
      def start_dag(self, dag_id, host):
          patch_string = f"http://{host}/api/v1/dags/{dag_id}"
          print(f"Calling [start_dag]#1 {patch_string}")
          result = self.session.patch(patch_string, json={"is_paused": False})
          try:
              result_json = result.json()
          except ValueError:
              result_json = str(result)
          print(f"Received [start_dag]#1 {result_json}")
  >       assert result.status_code == 200, f"Could not enable DAG: {result_json}"
  E       AssertionError: Could not enable DAG: {'detail': None, 'status': 404, 'title': "Dag with id: 'example_kubernetes_executor' not found", 'type': 'http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/stable/stable-rest-api-ref.html#section/Errors/NotFound'}
  E       assert 404 == 200
  E        +  where 404 = <Response [404]>.status_code
```

This can likely be improved but some additional checks before we start k8S tests that DAGs are already parsed and serialized.

--- a/kubernetes_tests/test_base.py
+++ b/kubernetes_tests/test_base.py
@@ -215,18 +215,6 @@ class BaseK8STest:
 
         # Maybe check if we can retrieve the logs, but then we need to extend the API
 
-    def _wait_until_dag_response(self, url, payload):
-        max_attempts = 10
-        result = {}
-        while max_attempts:
-            result = self.session.patch(url, json=payload)
-            if result.status_code == 200:
-                return result
-            else:
-                time.sleep(30)
-            max_attempts -= 1
-        return result
-
     def start_dag(self, dag_id, host):
         patch_string = f"http://{host}/api/v1/dags/{dag_id}"
         print(f"Calling [start_dag]#1 {patch_string}")

--- a/kubernetes_tests/test_base.py
+++ b/kubernetes_tests/test_base.py
@@ -215,10 +215,31 @@ class BaseK8STest:
 
         # Maybe check if we can retrieve the logs, but then we need to extend the API
 
+    def _wait_until_dag_response(self, url, payload):
+        max_attempts = 10
+        result = {}
+        while max_attempts:
+            result = self.session.patch(url, json=payload)
+            if result.status_code == 200:
+                return result
+            else:
+                time.sleep(30)
+            max_attempts -= 1
+        return result
+
     def start_dag(self, dag_id, host):
         patch_string = f"http://{host}/api/v1/dags/{dag_id}"
         print(f"Calling [start_dag]#1 {patch_string}")
-        result = self.session.patch(patch_string, json={"is_paused": False})
+        max_attempts = 10
+        result = {}
+        while max_attempts:
+            result = self.session.patch(patch_string, json={"is_paused": False})
+            if result.status_code == 200:
+                break
+
+            time.sleep(30)
+            max_attempts -= 1
+
         try:
             result_json = result.json()
         except ValueError:

